mesh/mesh_readers.cpp

```diff
@@ -886,215 +897,998 @@ void Mesh::ReadInlineMesh(std::istream &input, int generate_edges)
+      10,/* 10-node second order tetrahedron (4 nodes associated with the
+               vertices and 6 with the edges). */
+      27,/* 27-node second order hexahedron (8 nodes associated with the
+               vertices, 12 with the edges, 6 with the faces and 1 with
+               the volume). */
+      18,/* 18-node second order prism (6 nodes associated with the
+               vertices, 9 with the edges and 3 with the quadrangular
+               faces). */
+      14,/* 14-node second order pyramid (5 nodes associated with the
+               vertices, 8 with the edges and 1 with the quadrangular
+               face). */
+      1, // 1-node point.
+      8, /* 8-node second order quadrangle (4 nodes associated with the
+               vertices and 4 with the edges). */
+      20,/* 20-node second order hexahedron (8 nodes associated with the
+               vertices and 12 with the edges). */
+      15,/* 15-node second order prism (6 nodes associated with the
+               vertices and 9 with the edges). */
+      13,/* 13-node second order pyramid (5 nodes associated with the
+               vertices and 8 with the edges). */
+      9, /* 9-node third order incomplete triangle (3 nodes associated
+               with the vertices, 6 with the edges) */
+      10,/* 10-node third order triangle (3 nodes associated with the
+               vertices, 6 with the edges, 1 with the face) */
+      12,/* 12-node fourth order incomplete triangle (3 nodes associated
+               with the vertices, 9 with the edges) */
+      15,/* 15-node fourth order triangle (3 nodes associated with the
+               vertices, 9 with the edges, 3 with the face) */
+      15,/* 15-node fifth order incomplete triangle (3 nodes associated
+               with the vertices, 12 with the edges) */
+      21,/* 21-node fifth order complete triangle (3 nodes associated with
+               the vertices, 12 with the edges, 6 with the face) */
+      4, /* 4-node third order edge (2 nodes associated with the vertices,
+               2 internal to the edge) */
+      5, /* 5-node fourth order edge (2 nodes associated with the
+               vertices, 3 internal to the edge) */
+      6, /* 6-node fifth order edge (2 nodes associated with the vertices,
+               4 internal to the edge) */
+      20 /* 20-node third order tetrahedron (4 nodes associated with the
+               vertices, 12 with the edges, 4 with the faces) */
+   };
+
+
+
+void Mesh::ReadGmshBoundaryElements(std::vector<Element*> &elements_0D,
+                                    std::vector<Element*> &elements_1D,
+                                    std::vector<Element*> &elements_2D,
+                                    std::vector<Element*> &elements_3D)
 {
-   string buff;
-   double version;
-   int binary, dsize;
-   input >> version >> binary >> dsize;
-   if (version < 2.2)
+   if (!elements_3D.empty())
    {
-      MFEM_ABORT("Gmsh file version < 2.2");
+      Dim = 3;
```

Comment:
Do you know if GMesh supports surfaces meshes embeded in 3D.  If so this might need a little rework to support those.

Reply:
Yeah, this is what I was thinking based on our convo.  I need to do some research on both sides gmesh/mfem to see how these should be treated,